JavaScript/tests/syntax_test_typescript.ts

```diff
@@ -359,6 +359,26 @@ function f(this : any) {}
 //      ^^^ meta.type support.type.any
 //           ^^ storage.type.function.arrow
 
+    x ? (y) : z;
+//  ^ variable.other.readwrite
+//    ^ keyword.operator.ternary
+//      ^^^ meta.group
+//       ^ variable.other.readwrite
+//          ^ keyword.operator.ternary
+
+    x ? (y) : T => r : z;
+//  ^ variable.other.readwrite
+//    ^ keyword.operator.ternary
+//      ^^^^^^^^^^^^^ meta.function
+//      ^ meta.function.declaration
+//       ^ meta.binding.name variable.parameter.function
+//          ^ punctuation.separator.type
+//            ^ meta.type support.class
+//              ^^ storage.type.function.arrow
```

Comment:
Maybe `storage.type.function.arrow keyword.declaration.function.arrow` as per scope naming guidelines 
https://www.sublimetext.com/docs/scope_naming.html#keyword ?

Reply:
I don't have an opinion either way, but it's probably out of scope for this PR.

Reply:
Alright. You can go ahead & close this review.